.changeset/stupid-poets-grab.md

```diff
@@ -0,0 +1,21 @@
+---
+'@astrojs/markdoc': minor
+---
+
+Adds an "allowHTML" Markdoc integration option.
+
+When enabled, all Markdoc markup is processed via the htmlparser2 library to detect
+HTML elements and creating a modified set of markdown-it tokens which seamlessly interleave
+HTML markup elements with any Markdoc tags and nodes.
+
+This is a potential XSS vector as HTML in Markdoc markup will be rendered as real HTML nodes
+once this is enabled (it should be noted this is already the case for Markdown and MDX integrations
+as-is, by default.)
```

Comment:
```suggestion
When enabled, all HTML in Markdoc files will be processed, including HTML elements within Markdoc tags and nodes.
```

nit: No need for technical implementation details! I'd also suggest removing the XSS vector warning, since this nuance is better left to the README.